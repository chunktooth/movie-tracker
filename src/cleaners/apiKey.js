export const apiKey = '480eafc74b027454d89765098061cb64';

// jared - 480eafc74b027454d89765098061cb64
// Matt - 4eabf2c86624eba27b3bed0627180793
// om - e7f886548071b04cf8febc293d108511