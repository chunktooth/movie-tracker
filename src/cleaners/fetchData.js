import { apiKey } from './apiKeys';

const rootUrl = 'https://api.themoviedb.org/3';
const nowPlaying = `/movie/now_playing?page=1&language=en-US&api_key=${apiKey}&language=en-US`;

export const getNowPlaying = async () => {

  try {
    const response = await fetch(`${rootUrl}${nowPlaying}`);
    const nowPlayingData = await response.json();

    return nowPlayingData;
  } catch (error) {
    throw new Error ('its bad');
  }
};


export const addToFavorites = async (movie) => {

  try {
    const response = await fetch('/api/users/favorites/new', {
      method: 'POST',
      body: JSON.stringify(movie),
      headers: {
        'content-type': 'application/json'
      }
    });
    const favoriteData = await response.json();

    return favoriteData;
  } catch (error) {
    console.log(error);
  }
}

export const removeFromFavorites = async (userId, movieId) => {

  try {
    const response = await fetch(`/api/users/${userId}/favorites/${movieId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      }
    });
    const removedData = await response.json();
    
    return removedData;
  } catch (error) {
    console.log(error);
  }
}