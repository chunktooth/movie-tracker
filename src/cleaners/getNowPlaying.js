import { apiKey } from './apiKeys';

export const getNowPlaying = async () => {

  try {
    const response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?page=1&language=en-US&api_key=${apiKey}&language=en-US`);
    const nowPlayingData = await response.json();
    const cleanedMovies = cleanMovies(nowPlayingData)
    return cleanedMovies
  } catch (error) {
    throw new Error ('Cannot fetch');
  }
};

const cleanMovies = (movieData) => {
  return movieData.results.map(movie => ({
    movie_id: movie.id, 
    title: movie.title, 
    poster_path: movie.poster_path, 
    release_date: movie.release_date, 
    vote_average: movie.vote_average, 
    overview: movie.overview  
  }));
}