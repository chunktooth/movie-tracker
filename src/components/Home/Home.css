.home-now-playing {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}