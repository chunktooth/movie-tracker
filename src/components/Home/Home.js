import React from 'react';
import Card from '../Card/Card.js';
import './Home.css';


export const Home = ({ movies, favorites }) => {
  const displayMovies = movies.map((movie, index) => 
    {
      let selected = '';
      favorites.forEach(favorite => {
        if (favorite.movie_id === movie.movie_id) {
          selected = 'selected';
        }
      });

      return ( <Card 
        movieInfo={movie} 
        key={index}
        selected={selected}
      />
    );
  });

  return (
    <div className='row'>
      <div className='row__inner'>
        {displayMovies}
      </div>
    </div>
  );
};