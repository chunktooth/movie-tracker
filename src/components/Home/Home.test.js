import React from 'react';
import { Home, mapStateToProps } from './Home';
import { mockMovieProp } from '../../cleaners/mockData';
import { shallow } from 'enzyme';


describe('Home', () => {

  it('should match the snapshot', () => {
    let wrapper = shallow(<Home movies={mockMovieProp} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should map state correctly in store', () => {
    const mockStore = { movies: mockMovieProp };
    const mapped = mapStateToProps(mockStore);
    expect(mapped.movies).toEqual(mockStore.movies);
  });

});