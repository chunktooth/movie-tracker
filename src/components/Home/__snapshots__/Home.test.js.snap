// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    movies={
        Array [
            Object {
              "adult": "false",
              "backdrop_path": "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
              "genre_ids": "[28, 12, 14, 878]",
              "id": "284054",
              "original_language": "en",
              "original_title": "Black Panther",
              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
              "popularity": "249.862248",
              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "video": "false",
              "vote_average": "7.4",
              "vote_count": "3582",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          movieInfo={
                    Object {
                              "adult": "false",
                              "backdrop_path": "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
                              "genre_ids": "[28, 12, 14, 878]",
                              "id": "284054",
                              "original_language": "en",
                              "original_title": "Black Panther",
                              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                              "popularity": "249.862248",
                              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                              "release_date": "2018-02-13",
                              "title": "Black Panther",
                              "video": "false",
                              "vote_average": "7.4",
                              "vote_count": "3582",
                            }
          }
/>,
      ],
      "className": "home-now-playing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "movieInfo": Object {
            "adult": "false",
            "backdrop_path": "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
            "genre_ids": "[28, 12, 14, 878]",
            "id": "284054",
            "original_language": "en",
            "original_title": "Black Panther",
            "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
            "popularity": "249.862248",
            "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
            "release_date": "2018-02-13",
            "title": "Black Panther",
            "video": "false",
            "vote_average": "7.4",
            "vote_count": "3582",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            movieInfo={
                        Object {
                                    "adult": "false",
                                    "backdrop_path": "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
                                    "genre_ids": "[28, 12, 14, 878]",
                                    "id": "284054",
                                    "original_language": "en",
                                    "original_title": "Black Panther",
                                    "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                    "popularity": "249.862248",
                                    "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                    "release_date": "2018-02-13",
                                    "title": "Black Panther",
                                    "video": "false",
                                    "vote_average": "7.4",
                                    "vote_count": "3582",
                                  }
            }
/>,
        ],
        "className": "home-now-playing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "movieInfo": Object {
              "adult": "false",
              "backdrop_path": "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
              "genre_ids": "[28, 12, 14, 878]",
              "id": "284054",
              "original_language": "en",
              "original_title": "Black Panther",
              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
              "popularity": "249.862248",
              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "video": "false",
              "vote_average": "7.4",
              "vote_count": "3582",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
